airflow/providers/amazon/aws/hooks/quicksight.py

```diff
@@ -0,0 +1,145 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import time
+
+from botocore.exceptions import ClientError
+
+from airflow import AirflowException
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+from airflow.providers.amazon.aws.hooks.sts import StsHook
+
+
+class QuickSightHook(AwsBaseHook):
+    """
+    Interact with Amazon QuickSight.
+
+    Additional arguments (such as ``aws_conn_id``) may be specified and
+    are passed down to the underlying AwsBaseHook.
+    .. seealso::
+    :class:`~airflow.providers.amazon.aws.hooks.base_aws.AwsBaseHook`
+    """
+
+    NON_TERMINAL_STATES = {"INITIALIZED", "QUEUED", "RUNNING"}
+    FAILED_STATES = {"FAILED"}
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(client_type="quicksight", *args, **kwargs)
+
+    def create_ingestion(
+        self,
+        data_set_id: str,
+        ingestion_id: str,
+        ingestion_type: str,
+        wait_for_completion: bool = True,
+        check_interval: int = 30,
+    ):
+        """
+        Creates and starts a new SPICE ingestion for a dataset. Refreshes the SPICE datasets
+
+        :param data_set_id:  ID of the dataset used in the ingestion.
+        :param ingestion_id: ID for the ingestion.
+        :param ingestion_type: Type of ingestion . "INCREMENTAL_REFRESH"|"FULL_REFRESH"
+        :param wait_for_completion: if the program should keep running until job finishes
+        :param check_interval: the time interval in seconds which the operator
+            will check the status of QuickSight Ingestion
+        :return: Returns descriptive information about the created data ingestion
+            having Ingestion ARN, HTTP status, ingestion ID and ingestion status.
+        :rtype: Dict
+        """
+
+        self.log.info("Creating QuickSight Ingestion for data set id %s.", data_set_id)
+        quicksight_client = self.get_conn()
+        try:
+            sts_hook = StsHook()
```

Comment:
Do we want to create a hook every-time we call this method? Should not we use something like [cached_property](https://docs.python.org/3/library/functools.html#functools.cached_property)